packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -351,25 +385,31 @@ async function migrateMultiEnv(projectPath: string): Promise<void> {
     // version <= 2.4.1
     await fs.copy(path.join(fx, AppPackageFolderName), templateAppPackage);
   }
-  const targetManifestFile = path.join(templateAppPackage, MANIFEST_TEMPLATE);
-  await fs.rename(path.join(templateAppPackage, "manifest.source.json"), targetManifestFile);
-
+  const sourceManifestFile = path.join(templateAppPackage, REMOTE_MANIFEST);
+  const manifest: TeamsAppManifest = await readJson(sourceManifestFile);
+  await fs.remove(sourceManifestFile);
+  await moveIconsToResourceFolder(templateAppPackage, manifest);
   // generate manifest.remote.template.json
-  const manifest = await generateRemoteTemplate(targetManifestFile);
-  await fs.writeFile(targetManifestFile, JSON.stringify(manifest, null, 4));
-  await moveIconsToResourceFolder(templateAppPackage);
+  if (!migrateFromV1) {
```

Comment:
I wonder will the project migrated from v1 pass the check `needMigrateToArmAndMultiEnv`?
if not, we can remove `!migrateFromV1`

Reply:
now it can pass `needMigrateToArmAndMultiEnv`.  The migrate from v1 project should pass `needMigrateToArmAndMultiEnv`, it should send the same notification as normal v2 project